### ISSUE TITLE
[JavaWrappers] VideoCapture release() throws SIGKILL

### ISSUE THREAD

**Issue by AlexanderWolz** [opencv/opencv#7514]
##### System information (version)

- OpenCV => 3.1.0-master
- Operating System / Platform => macOS Sierra (10.12) x86_64
- Compiler => Apple LLVM version 8.0.0 (clang-800.0.38)

- build using Homebrew [today, 08:30CEST]: 
brew install ant
brew install opencv3 --HEAD --with-java --with-ffmpeg --with-tbb

##### Detailed description

If you use VideoCapture via Java wrappers and if you try to release the camera, it always crashes the Java process.  It can successfully read the frame though. Releasing the mat works fine, releasing the video capture crashes.

##### Steps to reproduce

```.java
	public static void main(String[] args) {
		VideoCapture capture = new VideoCapture(0);
		Mat mat = new Mat();
		capture.read(mat);
		mat.release();
		capture.release(); //crashes here
	}
```

Afterwards it shows following:

```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007fff954f9f6f, pid=4925, tid=0x000000000000080b
#
# JRE version: Java(TM) SE Runtime Environment (8.0_102-b14) (build 1.8.0_102-b14)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.102-b14 mixed mode bsd-amd64 compressed oops)
# Problematic frame:
# C  [libobjc.A.dylib+0x8f6f]  objc_release+0x1f
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
```

**Comment by alalek** [comment]
Could you try patch #7509?

**Comment by abratchik** [comment]
Probably duplicate of #7418

**Comment by AlexanderWolz** [comment]
Unfortunately, patch #7509 did not solve this issue.
downloaded master, patched file, and executed with following params:
```
cmake -D CMAKE_BUILD_TYPE=RELEASE -D CMAKE_INSTALL_PREFIX=/usr/local -D BUILD_SHARED_LIBS=OFF -D BUILD_PYTHON_SUPPORT=ON ..
```

**Comment by alalek** [comment]
@AlexanderWolz Thanks for report and check!
Let's track issue in #7418
